fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordContextProperties.java

```diff
@@ -97,4 +99,45 @@
      * Maximum number of segments to be merged at a time for auto-complete search with Lucene, during forceMerge for forceMergeDeletes.
      */
     public static final RecordLayerPropertyKey<Integer> LUCENE_AUTO_COMPLETE_MERGE_MAX_NUMBER = RecordLayerPropertyKey.integerPropertyKey("com.apple.foundationdb.record.lucene.autoCompleteMergeMaxNum", 2);
+
+    /**
+     * The index options for the Lucene field enabled with full-text search, that determines what information is stored in the index.
+     */
+    public static final RecordLayerPropertyKey<LuceneFieldIndexOptions> LUCENE_FULL_TEXT_FIELD_INDEX_OPTIONS = RecordLayerPropertyKey.enumPropertyKey("com.apple.foundationdb.record.lucene.fieldIndexOptions", LuceneFieldIndexOptions.DOCS_AND_FREQS_AND_POSITIONS);
+
+    /**
+     * Determines whether the Lucene field's index should also include term vectors.
+     */
+    public static final RecordLayerPropertyKey<Boolean> LUCENE_FULL_TEXT_WITH_TERM_VECTORS = RecordLayerPropertyKey.booleanPropertyKey("com.apple.foundationdb.record.lucene.withTermVectors", false);
+
+    /**
+     * Determines whether the positions are also stored in the term vectors of the Lucene field.
+     */
+    public static final RecordLayerPropertyKey<Boolean> LUCENE_FULL_TEXT_WITH_TERM_VECTOR_POSITIONS = RecordLayerPropertyKey.booleanPropertyKey("com.apple.foundationdb.record.lucene.withTermVectorPositions", false);
+
+    /**
+     * Determines whether to omit the normalization values for the Lucene field.
+     */
+    public static final RecordLayerPropertyKey<Boolean> LUCENE_FULL_TEXT_WITH_OMIT_NORMS = RecordLayerPropertyKey.booleanPropertyKey("com.apple.foundationdb.record.lucene.withOmitNorms", false);
+
+    /**
+     * Determines what information is stored in the index of the Lucene field.
+     */
+    public enum LuceneFieldIndexOptions {
+        DOCS,
+        DOCS_AND_FREQS,
+        DOCS_AND_FREQS_AND_POSITIONS,
+        DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS,
+    }
+
+    public static IndexOptions getIndexOptions(LuceneFieldIndexOptions value, Logger logger) {
```

Comment:
I'm not sure this is the right place for this translation of the Record Layer-defined `LuceneFieldIndexOptions` enum to the Lucene-defined `IndexOptions` enum, especially because it means that the property definition class now has a method that exposes something from under Lucene. It might make more sense to put it in the maintainer (or maybe even just to mark this function as `INTERNAL`).